src/Utils/time.ts

```diff
@@ -0,0 +1,55 @@
+import { metaphysics } from "./metaphysics"
+
+const getLocalTimestampInMilliSeconds = () => {
+  return Date.now()
+}
+
+interface SystemTimeResults {
+  data: {
+    system: {
+      time: {
+        unix: number
+      }
+    }
+  }
+}
+
+const fetchSystemTime = () => {
+  return metaphysics<SystemTimeResults>({
+    query: `
+      {
+        system {
+          time {
+            unix
+          }
+        }
+      }
+    `,
+  })
+}
+
+const getGravityTimestampInMilliSeconds = async () => {
+  const startTime = getLocalTimestampInMilliSeconds()
+  const { data } = await fetchSystemTime()
+
+  const possibleNetworkLatencyInMilliSeconds =
+    (getLocalTimestampInMilliSeconds() - startTime) / 2
+  const serverTimestampInMilliSeconds =
+    data.system.time.unix * 1e3 + possibleNetworkLatencyInMilliSeconds
+
+  return serverTimestampInMilliSeconds
+}
+
+export const getOffsetBetweenGravityClock = async () => {
+  try {
+    const gravityClock = await getGravityTimestampInMilliSeconds()
```

Comment:
Might be wise to measure how long this request takes and discard any that take longer than a couple of seconds or so. Depends how accurate we want to be.